OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -263,7 +263,7 @@ public void onEventMainThread(DiscussionThreadUpdatedEvent event) {
     @SuppressWarnings("unused")
     public void onEventMainThread(DiscussionCommentPostedEvent event) {
         // If a new response/comment was posted in a listed thread, we need to update the list
-        for (int i = 0; i < discussionPostsAdapter.getCount(); ++i) {
+        for (int i = 0; i < discussionPostsAdapter.getItemCount(); ++i) {
             final DiscussionThread discussionThread = discussionPostsAdapter.getItem(i);
             if (discussionThread.containsComment(event.getComment())) {
                 // No need to update the discussionThread object because its already updated on
```

Comment:
Please use 
`discussionPostsAdapter.notifyItemRangeChanged(0, i);` instead of `discussionPostsAdapter.notifyDataSetChanged();` here for optimization.